tasks/podman_pods.yml

```diff
@@ -0,0 +1,32 @@
+---
+
+- name: Determine list of required pods and ports
+  when: service.pod is defined
+  loop: "{{ podman_services }}"
+  loop_control:
+    loop_var: service
+  set_fact:
+    podman_pods: "{{ podman_pods | default({}) | combine({service.pod.name: {'name': service.pod.name, 'ports': service.pod.ports | default([]) + (podman_pods[service.pod.name]['ports'] | default([]))}}) }}"
+
+- name: Delete pods with missing port bindings
```

Comment:
The reason I never added pod support was because it would require some gnarly use of the shell module like this.

I keep waiting for upstream Ansible to a module like `podman_pod` support but that may never happen.

I have two real requests when it comes to this:

1. What does `--pod` provide over `--network`?

2. If this is to get merged it will need a way to be idempotent. As it stands if I utilize the `pod` option for any of my services this will always report `changed`. This breaks one of the core promises of Configuration Management.

Point number 2 is why I never did this, you can do some wicked things with `args` and shell like:

```
shell: echo "something" > this_file
args:
    creates: this_file
```

To make it "idempotent"-ish and I think I'd be willing to accept that but this is already a pretty gnarly script.

Reply:
I anticipated that it could be problematic to put both changes in one PR and to implement the pod feature in that way. For my use case (Apache Guacamole) I needed an automation to connect 3 containers and publish some of their ports. Neither using pods nor network was possible with the existing Ansible modules, so I picked the pod option.

I don't think we can change the pod feature to have the quality (i.e. idempotency) you require for this project. While we wait for a native Ansible module to cover this feature I'll keep the branch published in case somebody needs to configure pods.

I split the pod feature into #2. Also you can close this PR.

Thanks for your comments and excellent work!

Reply:
I had a similar problem when trying to deploy Nextcloud without pods I eventually gave up in the end and just exposed all the ports for everything and networked them the old-school way.

I think it's a very valid use case and I would love to see this feature through. With the new Ansible split into modules I'm not even sure where to report this upstream.

I also do not know if we can ship custom modules with roles on Galaxy if so that might be an option?

Perhaps an alternative is to do a little more abusing of the shell module. I've also attempted to make pods be [https://fedoramagazine.org/systemd-template-unit-files/](SystemD template units) in the past which almost worked but I didn't need pods bad enough to push it over the line.

Reply:
Shipping custom modules on Galaxy is made possible by [Collections](https://docs.ansible.com/ansible/latest/user_guide/collections_using.html).

I just stumbled upon this collection on Galaxy: https://github.com/containers/ansible-podman-collections. The project provides additional modules for pods and networks. However it seems that it doesn't let you create networks.

Reply:
Perusing the source code I found https://github.com/containers/ansible-podman-collections/blob/master/plugins/modules/podman_network.py

Which does appear to let us create networks. I bet we could pull this as a dependency and make it work.

Reply:
Hi @chasinglogic @jwhb ,

I'm really interested in this feature. Do you think this is something you can work on ?

Best,
Jerome

Reply:
Hey @jrevillard,

I'm not sure when I can get to working on this